lib/srv/monitor_test.go

```diff
@@ -191,3 +192,26 @@ func TestMonitorDisconnectExpiredCertBeforeTimeNow(t *testing.T) {
 		t.Fatal("Client is still connected.")
 	}
 }
+
+func TestTrackingReadConnEOF(t *testing.T) {
+	server, client := net.Pipe()
+	defer client.Close()
+
+	// Close the server to force client reads to instantly return EOF.
+	require.NoError(t, server.Close())
+
+	// Wrap the client in a TrackingReadConn.
+	ctx, cancel := context.WithCancel(context.Background())
+	tc, err := NewTrackingReadConn(TrackingReadConnConfig{
+		Conn:    client,
+		Clock:   clockwork.NewFakeClock(),
+		Context: ctx,
+		Cancel:  cancel,
+	})
+	require.NoError(t, err)
+
+	// Make sure it returns an EOF and not a wrapped exception.
+	buf := make([]byte, 64)
+	_, err = tc.Read(buf)
+	require.True(t, err == io.EOF)
```

Comment:
```suggestion
	require.Equal(t, io.EOF, err)
```

This assert will print a much nicer error message if it fails.

The way you have it written here, the only error you'll get is "expected true, got false".